src/package-install-scripts.js

```diff
@@ -280,15 +292,55 @@ export default class PackageInstallScripts {
 
     await Promise.all(workers);
 
-    // cache all build artifacts
-    for (const pkg of pkgs) {
-      if (this.packageCanBeInstalled(pkg)) {
-        const ref = pkg._reference;
-        invariant(ref, 'expected reference');
-        const loc = this.config.generateHardModulePath(ref);
-        const beforeFiles = beforeFilesMap.get(loc);
-        invariant(beforeFiles, 'files before installation should always be recorded');
-        await this.saveBuildArtifacts(loc, pkg, beforeFiles, set.spinners[0]);
+    // generate built package as prebuilt one for offline mirror
+    if (this.config.packBuiltPackages) {
+      for (const pkg of pkgs) {
+        if (this.packageCanBeInstalled(pkg)) {
+          const offlineMirrorPath = this.config.getOfflineMirrorPath();
+          if (!offlineMirrorPath) {
+            break;
```

Comment:
If the packBuiltPackages option is set but no offline mirror has been set, the `saveBuildArtifacts` function won't be called. Is it expected?

Reply:
Good point!
Not inteded